### PR TITLE
feat(eth): introduce output type for passing full event msgs

### PR DESCRIPTION
**Pull request by mattevans**
👋 Hey team, it was great to meet some of you at DC7.

Xatu consumes Hermes events via the `DataStreamTypeCallback` and would really benefit in having the entire event payload passed-through. This PR introduces the concept of an `DataStreamOutputType` and implements `Kinesis` and `Full` renderers for each.

- Kinesis event structure/output is unchanged. 
- StdLogger event structure/output is unchanged. 
- Callback event structure/output does change; using qualified types over `map[string]any` for its event payload data.

<details>
  <summary>Running Locally</summary>

  ```bash
  go run ./cmd/hermes --data.stream.type=callback eth \
	--chain=devnet \
	--devp2p.host=0.0.0.0 \
	--libp2p.host=0.0.0.0 \
	--prysm.host=127.0.0.1 \
	--prysm.port.http=33001 \
	--prysm.port.grpc=33003 \
	--genesis.ssz.url=http://127.0.0.1:40000/network-configs/genesis.ssz \
	--config.yaml.url=http://127.0.0.1:40000/network-configs/config.yaml \
	--bootnodes.yaml.url=http://127.0.0.1:40000/network-configs/boot_enr.yaml \
	--deposit-contract-block.txt.url=http://127.0.0.1:40000/network-configs/deposit_contract_block.txt
  ```
</details>

### CODE REVIEW

File: host/flush_tracer.go
Comment (cortze) [inline host/flush_tracer.go:21]: Is it necessary to have the topic field also in the Trace event? (doesn't hurt but feels duplicated)
The extended Payload should already have the Topic of the message. Are you guys going to need it for some easier filter afterwards?

**Review by cortze — APPROVED** [review]
Thanks @mattevans for the PR. overall,  LGTM, nice and clean. I just have one peaky comment that could even be ignored.

**Review by dennis-tra — APPROVED** [review]

File: host/flush_tracer.go
Comment (mattevans) [reply host/flush_tracer.go:21]: > Are you guys going to need it for some easier filter afterwards?

Yeah bang on! 💯 Xatu receives the `*host.TraceEvent` and uses the `Topic` to determine what the extended payload data is, to then cast into a more concrete type, eg: `TraceEventAttestation`, `TraceEventVoluntaryExit`, etc.

> doesn't hurt but feels duplicated

Agree, I was a bit hesitant to add it, but couldn't find another way that wasn't really intrusive.

**Comment by mattevans** [comment]
> Just one suggestion (not blocking the merge), the `DataStream` interface could expose the output renderer directly

Thanks 🙌 

I _think_ if I did this, it would mean the `host` pkg then has a dep on the `eth` pkg, which I was trying to avoid. 🤔 

---

Really appreciate the such prompt reviews @dennis-tra @cortze. It looks like there's a few workflows awaiting approval before it can be merged. I don't have perms to kick them off.

**Comment by dennis-tra** [comment]
> I think if I did this, it would mean the host pkg then has a dep on the eth pkg, which I was trying to avoid. 🤔

Oh, of course! Yeah, let's avoid that. Thanks for pointing this out 👍

**Pull request merged**
